File: java-algo/src/main/java/com/leetcode/Problem704.java
package com.leetcode;

/**
 * 704. Binary Search
 * https://leetcode.com/problems/binary-search/
 */
public class Problem704 {
    public int search(int[] nums, int target) {
        int i = 0, j = nums.length - 1, mid;
        while (i <= j) {
            mid = (j - i) / 2 + i;
            if (nums[mid] == target)
                return mid;
            if (nums[mid] < target)
                i = mid + 1;
            else
                j = mid - 1;
        }
        return -1;
    }
}

File: python-algo/leetcode/problem_704.py
"""704. Binary Search
https://leetcode.com/problems/binary-search/
"""
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums) - 1
        while i < j:
            mid = (i + j) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                i = mid + 1
            else:
                j = mid - 1
        return i if nums[i] == target else -1
